data-prep-pipeline/scripts/subtype_tree/create_context_matrix.py

```
@@ -0,0 +1,244 @@
+    break any assumptions/rules that are used by the portal
+
+    Inputs:
+        - subtype_tree (pandas df): The subtype tree as created by create_subtype_tree.py
+        - context_matrix (pandas df): The final result of the context matrix
+
+    Outputs: None (function will raise an error if any assert fails)
+    '''
+
+    #assert that all column names are unique
+    assert(len(context_matrix.columns) == len(context_matrix.columns.unique()))
+
+    #assert that all nodes in the subtype tree have a corresponding column
+    cm_nodes = set(context_matrix.columns)
+    st_nodes = set(subtype_tree.DepmapModelType.dropna())\
+                .union(set(subtype_tree.MolecularSubtypeCode.dropna()))
+
+    assert(set.intersection(cm_nodes, st_nodes) == set.union(cm_nodes, st_nodes))
+
+def create_context_matrix(
+        model_taiga_id,
+        molecular_subtypes_taiga_id,
+        subtype_tree_path
+    ):
+    model_tree, subtype_tree, genetic_subtypes = load_data(
+        model_taiga_id,
+        molecular_subtypes_taiga_id,
+        subtype_tree_path
+    )
+
+    context_matrix = construct_matrix(
+        model_tree,
+        subtype_tree,
+        genetic_subtypes
+    )
+
+    sanity_check_results(subtype_tree, context_matrix)
+
+    return context_matrix
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Create ContextMatrix"
+    )
+    parser.add_argument("model", help="Taiga ID of model table")
+    parser.add_argument("molecular_subtypes", help="Taiga ID of Omics Inferred Molecular Subtypes")
+    parser.add_argument("subtype_tree", help="Filepath for the SubtypeTree")
+    parser.add_argument("output", help="filepath to write the output")
+    args = parser.parse_args()
+
+    context_matrix = create_context_matrix(
+        args.model,
+        args.molecular_subtypes,
+        args.subtype_tree
+    )
+
+    if context_matrix is not None:
```

Is there ever a case where the context matrix being None isn't a gigantic problem? If None is not an acceptable result, should we just let this blow up here so we're aware sooner?